lib/spack/spack/test/cmd/dev_build.py

```diff
@@ -147,3 +161,222 @@ def test_dev_build_fails_nonexistent_package_name(mock_packages):
 def test_dev_build_fails_no_version(mock_packages):
     output = dev_build('dev-build-test-install', fail_on_error=False)
     assert 'dev-build spec must have a single, concrete version' in output
+
+
+def test_dev_build_env(tmpdir, mock_packages, install_mockery,
+                       mutable_mock_env_path):
+    # setup dev-build-test-install package for dev build
+    # we can concretize outside environment because it has no dev-build deps
```

Comment:
I thought it was supposed to be possible to concretize with dev-build deps outside of an environment (https://github.com/spack/spack/pull/15256#discussion_r492322152). Or am I missing something?

Reply:
If `foo ^bar` and `bar` is configured as a dev build in environment `baz`, if we concretize `foo` outside of `baz` we'll get a spec of `foo` depending on a spec of `bar` that is not a dev build. If we concretize it in `baz`, it will concretize to a spec that satisfies `foo ^bar dev_path=/path/configured/in/baz`.

Reply:
OK I think the comment needs to be rewritten. This doesn't sound like an issue of what can and cannot possibly happen (e.g. whether doing the install will result in an error), but rather the expected behavior (I think the idea is that the environment would make sure that `foo` concretizes with `bar dev_path=...` if you had built `bar dev_path=...` in the environment, right?).

Reply:
Yes, it's an issue of expected behavior, and it's comparing it to other tests for which we couldn't concretize outside the environment (without causing unexpected behavior

Reply:
I'll update the comment

Reply:
I wonder whether github changed the version of the test software under the hood and it's more particular than it used to be?